server/src/main/scala/actors/queues/FlightsRouterActor.scala

```diff
@@ -82,35 +83,26 @@ object FlightsRouterActor {
         .pipeTo(replyTo)
     }
 
-  def runAndCombine(source: Future[Source[FlightsWithSplits, NotUsed]])(implicit mat: ActorMaterializer, ec: ExecutionContext): Future[FlightsWithSplits] = source
-    .flatMap(
-      _.runWith(Sink.reduce[FlightsWithSplits](_ ++ _))
+  def runAndCombine(eventualSource: Future[Source[FlightsWithSplits, NotUsed]])
+                   (implicit mat: ActorMaterializer, ec: ExecutionContext): Future[FlightsWithSplits] = eventualSource
+    .flatMap(source => source
+      .log(getClass.getName)
+      .runWith(Sink.reduce[FlightsWithSplits](_ ++ _))
     )
 }
 
-class FlightsRouterActor(
-                          updatesSubscriber: ActorRef,
-                          terminals: Iterable[Terminal],
-                          flightsByDayLookup: FlightsLookup,
-                          updateFlights: FlightsUpdate
-                        ) extends Actor with ActorLogging {
-
-  implicit val dispatcher: ExecutionContextExecutor = context.dispatcher
-  implicit val mat: ActorMaterializer = ActorMaterializer.create(context)
-  implicit val timeout: Timeout = new Timeout(60 seconds)
+class FlightsRouterActor(updatesSubscribers: ActorRef,
+                         terminals: Iterable[Terminal],
+                         flightsByDayLookup: FlightsLookup,
+                         updateFlights: FlightsUpdate
+                        ) extends RouterActorLike[FlightUpdates, (Terminal, UtcDate)] {
+  override var maybeUpdatesSubscriber: Option[ActorRef] = Option(updatesSubscribers)
```

Comment:
is this optional?

Reply:
it's from RouterActorLike, which has to make it optional so that it can be sent the subscriber after it's been new'd up

Reply:
I'm not crazy about this pattern - shouldn't we make sure the subscriber is ready before we create the routerActor? We're allowing the actor to be in an invalid state by making it optional.

Reply:
Yeah I was waiting for you to complain about this pattern again ;) I'll have another look to see if there's a way around it. Now a bunch of complexity has been removed it may not have to be this way any longer